### PR TITLE
Student Grouping Code

### PR DESCRIPTION
**Pull request by aadiraju**
Runs student grouping and performs performance analysis per activity type using sample research questions for both impressions survey, and background survey.

### CODE REVIEW

File: scripts/student_grouping/student_grouping.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+from pandas import DataFrame
+
+
+def categorize(x, threshold):
+    """
+    Uses the threshold value to map x onto [-1, 0, 1]
```
Comment (Opeyem1a) [inline scripts/student_grouping/student_grouping.py:9]: So both sides of the threshold include the threshold itself? We should put that in the docstring/report/etc

File: scripts/student_grouping/student_grouping.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    else:
+        return 0
+
+
+def group_students(survey_path, facets, threshold) -> DataFrame:
+    """
+    Groups students based on the facet scores interpolated from the surveys into positive/neutral/negative groups and
+    displays a histogram for each facet.
+
+    :param threshold: The boundaries for the neutral group: [-threshold, +threshold]
```
Comment (Opeyem1a) [inline scripts/student_grouping/student_grouping.py:28]: Based on the comment above's resolution, resolve this

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

**Review by nononovia — DISMISSED** [review]
lgtm, the changes Opey mentioned seems to be resolved already?

**Comment by aadiraju** [comment]
@Opeyem1a does this look good to you?

File: scripts/student_grouping/student_grouping.py (hunk 2 above)
Comment (aadiraju) [reply scripts/student_grouping/student_grouping.py:28]: The set notation [] indicates inclusivity on both ends

**Review by Opeyem1a — APPROVED** [review]

**Pull request merged**